sdk/src/main/scala/io/horizen/account/state/StateDbAccountStateView.scala

```diff
@@ -154,15 +154,14 @@ class StateDbAccountStateView(
     })
   }
 
-  def getOrderedForgingStakesInfoSeq(epochNumber: Int): Seq[ForgingStakeInfo] = {
-    val isForkV1_4Active = Version1_4_0Fork.get(epochNumber).active
-    val minStakeFilter: ForgingStakeInfo => Boolean = if (isForkV1_4Active) {
+  def getOrderedForgingStakesInfoSeq(epochNumber: Int, stakeEpochNumber: Int): Seq[ForgingStakeInfo] = {
+    val minStakeFilter: ForgingStakeInfo => Boolean = if (Version1_4_0Fork.get(stakeEpochNumber).active) {
       fsi => fsi.stakeAmount >= minForgerStake
     } else {
       _ => true
     }
 
-    if (isForkV1_4Active && forgerStakesV2Provider.isActive(this)) {
+    if (Version1_4_0Fork.get(epochNumber).active && forgerStakesV2Provider.isActive(this)) {
```

Comment:
I think here stakeEpochNumber should be used instead of epochNumber, because this check is used to decide if stake V2 or stake V1 should be used. V2 requires to have in the statedb the activation flag set, but the statedb is the one related to stakeEpochNumber, not epochNumber.
However, I think that maybe just one epoch param could be used, that is the stakeEpoch when the block is forged, while it is the block epoch when the block is applied to the state.
In fact, fork 1.4 requires stakeEpoch, but also fork 1.3 requires the same for the exact same reason. Fork 1.3 is used for deciding if, for forger stake msg processor V1, V1 or V2 storage model should be used and, also in this case, at the end there is a flag in the statedb. So, in the end, the epoch of the statedb trumps the epoch number passed as param.